File: android/src/main/java/io/openim/flutter_openim_sdk/manager/SignalingManager.java
package io.openim.flutter_openim_sdk.manager;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.openim.flutter_openim_sdk.listener.OnBaseListener;
import io.openim.flutter_openim_sdk.listener.OnSignalingListener;
import open_im_sdk.Open_im_sdk;

public class SignalingManager extends BaseManager {

    public void setSignalingListener(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.setSignalingListener(new OnSignalingListener());

        result.success(null);
    }

    public void signalingInvite(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.signalingInvite(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                jsonValue(methodCall, "signalingInfo"));
    }

    public void signalingInviteInGroup(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.signalingInviteInGroup(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                jsonValue(methodCall, "signalingInfo"));
    }

    public void signalingAccept(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.signalingAccept(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                jsonValue(methodCall, "signalingInfo"));
    }

    public void signalingReject(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.signalingReject(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                jsonValue(methodCall, "signalingInfo"));
    }

    public void signalingCancel(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.signalingCancel(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                jsonValue(methodCall, "signalingInfo"));
    }

    public void signalingHungUp(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.signalingHungUp(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                jsonValue(methodCall, "signalingInfo"));
    }

    public void signalingGetRoomByGroupID(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.signalingGetRoomByGroupID(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "groupID"));
    }

    public void signalingGetTokenByRoomID(MethodCall methodCall, MethodChannel.Result result) {
        Open_im_sdk.signalingGetTokenByRoomID(
                new OnBaseListener(result, methodCall),
                value(methodCall, "operationID"),
                value(methodCall, "roomID"));
    }

//    public void signalingUpdateMeetingInfo(MethodCall methodCall, MethodChannel.Result result) {
//        Open_im_sdk.signalingUpdateMeetingInfo(
//                new OnBaseListener(result, methodCall),
//                value(methodCall, "operationID"),
//                jsonValue(methodCall, "info"));
//    }
//
//    public void signalingCreateMeeting(MethodCall methodCall, MethodChannel.Result result) {
//        Open_im_sdk.signalingCreateMeeting(
//                new OnBaseListener(result, methodCall),
//                value(methodCall, "operationID"),
//                jsonValue(methodCall, "info"));
//    }
//
//    public void signalingJoinMeeting(MethodCall methodCall, MethodChannel.Result result) {
//        Open_im_sdk.signalingJoinMeeting(
//                new OnBaseListener(result, methodCall),
//                value(methodCall, "operationID"),
//                jsonValue(methodCall, "info"));
//    }
//
//    public void signalingOperateStream(MethodCall methodCall, MethodChannel.Result result) {
//        Open_im_sdk.signalingOperateStream(
//                new OnBaseListener(result, methodCall),
//                value(methodCall, "operationID"),
//                value(methodCall, "streamType"),
//                value(methodCall, "roomID"),
//                value(methodCall, "userID"),
//                value(methodCall, "mute"),
//                value(methodCall, "muteAll"));
//    }
//
//    public void signalingGetMeetings(MethodCall methodCall, MethodChannel.Result result) {
//        Open_im_sdk.signalingGetMeetings(
//                new OnBaseListener(result, methodCall),
//                value(methodCall, "operationID"));
//    }
//
//    public void signalingCloseRoom(MethodCall methodCall, MethodChannel.Result result) {
//        Open_im_sdk.signalingCloseRoom(
//                new OnBaseListener(result, methodCall),
//                value(methodCall, "operationID"),
//                value(methodCall, "roomID"));
//    }
//
//    public void signalingSendCustomSignal(MethodCall methodCall, MethodChannel.Result result) {
//        Open_im_sdk.signalingSendCustomSignal(
//                new OnBaseListener(result, methodCall),
//                value(methodCall, "operationID"),
//                value(methodCall, "customInfo"),
//                value(methodCall, "roomID"));
//    }
}

File: ios/Classes/Module/SignalingManager.swift
import Foundation
import OpenIMCore

public class SignalingManager: BaseServiceManager {
    
    public override func registerHandlers() {
        super.registerHandlers()
        self["setSignalingListener"] = setSignalingListener
        self["signalingInvite"] = signalingInvite
        self["signalingInviteInGroup"] = signalingInviteInGroup
        self["signalingAccept"] = signalingAccept
        self["signalingReject"] = signalingReject
        self["signalingCancel"] = signalingCancel
        self["signalingHungUp"] = signalingHungUp
        self["signalingGetRoomByGroupID"] = signalingGetRoomByGroupID
        self["signalingGetTokenByRoomID"] = signalingGetTokenByRoomID
        self["signalingUpdateMeetingInfo"] = signalingUpdateMeetingInfo
        self["signalingCreateMeeting"] = signalingCreateMeeting
        self["signalingJoinMeeting"] = signalingJoinMeeting
        self["signalingOperateStream"] = signalingOperateStream
        self["signalingGetMeetings"] = signalingGetMeetings
        self["signalingCloseRoom"] = signalingCloseRoom
    }

    func setSignalingListener(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSetSignalingListener(SignalingListener(channel: channel))
        callBack(result)
    }

    func signalingInvite(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingInvite(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[jsonString: "signalingInfo"])
    }
    
    func signalingInviteInGroup(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingInviteInGroup(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[jsonString: "signalingInfo"])
    }
    
    func signalingAccept(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingAccept(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[jsonString: "signalingInfo"])
    }
    
    func signalingReject(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingReject(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[jsonString: "signalingInfo"])
    }
    
    func signalingCancel(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingCancel(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[jsonString: "signalingInfo"])
    }

    func signalingHungUp(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingHungUp(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[jsonString: "signalingInfo"])
    }
    
    func signalingGetRoomByGroupID(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingGetRoomByGroupID(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "groupID"])
    }
    
    func signalingGetTokenByRoomID(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingGetTokenByRoomID(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "roomID"])
    }
    
    func signalingUpdateMeetingInfo(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingUpdateMeetingInfo(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[jsonString: "info"])
    }
    
    func signalingCreateMeeting(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingCreateMeeting(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[jsonString: "info"])
    }
    
    func signalingJoinMeeting(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingJoinMeeting(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[jsonString: "info"])
    }
    
    func signalingOperateStream(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingOperateStream(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "streamType"], methodCall[string: "roomID"], methodCall[string: "userID"], methodCall[bool: "mute"], methodCall[bool: "muteAll"])
    }
    
    func signalingGetMeetings(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingGetMeetings(BaseCallback(result: result), methodCall[string: "operationID"])
    }
    
    func signalingCloseRoom(methodCall: FlutterMethodCall, result: @escaping FlutterResult){
        Open_im_sdkSignalingCloseRoom(BaseCallback(result: result), methodCall[string: "operationID"], methodCall[string: "roomID"])
    }
}
public class SignalingListener: NSObject, Open_im_sdk_callbackOnSignalingListenerProtocol {
  
    private let channel:FlutterMethodChannel
    
    init(channel:FlutterMethodChannel) {
        self.channel = channel
    }
    
    public func onInvitationCancelled(_ s: String?) {
        CommonUtil.emitEvent(channel: channel, method: "signalingListener", type: "onInvitationCancelled", errCode: nil, errMsg: nil, data: s)
    }
    
    public func onInvitationTimeout(_ s: String?) {
        CommonUtil.emitEvent(channel: channel, method: "signalingListener", type: "onInvitationTimeout", errCode: nil, errMsg: nil, data: s)
    }
    
    public func onInviteeAccepted(_ s: String?) {
        CommonUtil.emitEvent(channel: channel, method: "signalingListener", type: "onInviteeAccepted", errCode: nil, errMsg: nil, data: s)
    }
    
    public func onInviteeAccepted(byOtherDevice s: String?) {
        CommonUtil.emitEvent(channel: channel, method: "signalingListener", type: "onInviteeAcceptedByOtherDevice", errCode: nil, errMsg: nil, data: s)
    }
    
    public func onInviteeRejected(_ s: String?) {
        CommonUtil.emitEvent(channel: channel, method: "signalingListener", type: "onInviteeRejected", errCode: nil, errMsg: nil, data: s)
    }
    
    public func onInviteeRejected(byOtherDevice s: String?) {
        CommonUtil.emitEvent(channel: channel, method: "signalingListener", type: "onInviteeRejectedByOtherDevice", errCode: nil, errMsg: nil, data: s)
    }
    
    public func onReceiveNewInvitation(_ s: String?) {
        CommonUtil.emitEvent(channel: channel, method: "signalingListener", type: "onReceiveNewInvitation", errCode: nil, errMsg: nil, data: s)
    }
    
    public func onHangUp(_ s: String?) {
        CommonUtil.emitEvent(channel: channel, method: "signalingListener", type: "onHangUp", errCode: nil, errMsg: nil, data: s)
    }
    
    public func onRoomParticipantConnected(_ s: String?) {
        CommonUtil.emitEvent(channel: channel, method: "signalingListener", type: "onRoomParticipantConnected", errCode: nil, errMsg: nil, data: s)
    }
    
    public func onRoomParticipantDisconnected(_ s: String?) {
        CommonUtil.emitEvent(channel: channel, method: "signalingListener", type: "onRoomParticipantDisconnected", errCode: nil, errMsg: nil, data: s)
    }
    
    public func onStreamChange(_ s: String?) {
        CommonUtil.emitEvent(channel: channel, method: "signalingListener", type: "onStreamChange", errCode: nil, errMsg: nil, data: s)
    }
    
}
